src/hotspot/share/utilities/nativeCallStack.hpp

```diff
@@ -54,6 +54,7 @@
 class MemTracker;
 
 class NativeCallStack : public StackObj {
+  friend class VMATreeTest;
```

Comment:
I am surprised friend is needed, the private section of this class being so tiny. What does friend give you what you could not get via normal accessors?

Reply:
This permits access to the `_tree` field, so we can inspect it to check that it conforms to the shape we expect it to be. I don't think that we should have an accessor to that field, as that forms a part of its public interface.

Test fixtures being friend classes is imo not a code smell, but a natural consequence of doing clear box testing.